internal/bgp/frrk8s/frrk8s.go

```diff
@@ -0,0 +1,380 @@
+			neighbor = frrv1beta1.Neighbor{
+				ASN:           s.PeerASN,
+				Address:       host,
+				Port:          &portUint16,
+				HoldTime:      &metav1.Duration{Duration: s.HoldTime},
+				KeepaliveTime: &metav1.Duration{Duration: s.KeepAliveTime},
+				BFDProfile:    s.BFDProfile,
+				EBGPMultiHop:  s.EBGPMultiHop,
+				ToAdvertise: frrv1beta1.Advertise{
+					Allowed: frrv1beta1.AllowedOutPrefixes{
+						Prefixes: make([]string, 0),
+					},
+					PrefixesWithLocalPref: make([]frrv1beta1.LocalPrefPrefixes, 0),
+					PrefixesWithCommunity: make([]frrv1beta1.CommunityPrefixes, 0),
+				},
+				// TODO password
+			}
+		}
+
+		/* As 'session.advertised' is a map, we can be sure there are no
+		   duplicate prefixes and can, therefore, just add them to the
+		   'neighbor.Advertisements' list. */
+		prefixesForCommunity := map[string][]string{}
+		prefixesForLocalPref := map[uint32][]string{}
+		for _, adv := range s.advertised {
+			if !adv.MatchesPeer(s.SessionName) {
+				continue
+			}
+
+			prefix := adv.Prefix.String()
+			neighbor.ToAdvertise.Allowed.Prefixes = append(neighbor.ToAdvertise.Allowed.Prefixes, prefix)
+			rout.prefixes[prefix] = prefix
+
+			for _, c := range adv.Communities {
+				community := c.String()
+				prefixesForCommunity[community] = append(prefixesForCommunity[community], prefix)
+			}
+			if adv.LocalPref != 0 {
+				prefixesForLocalPref[adv.LocalPref] = append(prefixesForLocalPref[adv.LocalPref], prefix)
+			}
+		}
+		neighbor.ToAdvertise.PrefixesWithCommunity = toAdvertiseWithCommunity(prefixesForCommunity)
+		neighbor.ToAdvertise.PrefixesWithLocalPref = toAdvertiseWithLocalPref(prefixesForLocalPref)
+		sort.Slice(neighbor.ToAdvertise.Allowed.Prefixes, func(i, j int) bool {
+			return neighbor.ToAdvertise.Allowed.Prefixes[i] > neighbor.ToAdvertise.Allowed.Prefixes[j]
+		})
+
+		rout.neighbors[neighborName] = neighbor
+	}
+
+	for _, r := range sortMap(routers) {
+		toAdd := frrv1beta1.Router{
+			ASN:       r.myASN,
+			ID:        r.routerID,
+			VRF:       r.vrf,
+			Neighbors: sortMap(r.neighbors),
+			Prefixes:  sortMap(r.prefixes),
+		}
+		newConfig.Spec.BGP.Routers = append(newConfig.Spec.BGP.Routers, toAdd)
+	}
+
+	for _, bfd := range sm.bfdProfiles {
+		bfd := bfd
+		toAdd := frrv1beta1.BFDProfile{
+			Name: bfd.Name,
+		}
+		if bfd.ReceiveInterval != nil {
+			toAdd.ReceiveInterval = bfd.ReceiveInterval
+		}
+		if bfd.TransmitInterval != nil {
+			toAdd.TransmitInterval = bfd.TransmitInterval
+		}
+		if bfd.DetectMultiplier != nil {
+			toAdd.TransmitInterval = bfd.TransmitInterval
+		}
+		if bfd.EchoInterval != nil {
+			toAdd.EchoInterval = bfd.EchoInterval
+		}
+		if bfd.MinimumTTL != nil {
+			toAdd.MinimumTTL = bfd.MinimumTTL
+		}
+		toAdd.EchoMode = &bfd.EchoMode
+		toAdd.PassiveMode = &bfd.PassiveMode
+
+		newConfig.Spec.BGP.BFDProfiles = append(newConfig.Spec.BGP.BFDProfiles, toAdd)
+	}
+	sm.desiredConfig = newConfig
+	toNotify := sm.desiredConfig.DeepCopy()
+	sm.configChangedCallback(*toNotify)
```

Comment:
I thought about that but I also thought that we have the debouncer in frrk8s, plus some squashing will be introduced because of the api delays. I'd say let's try without and see if that fits the bill.

Reply:
fair enough, I'll add a debouncer to the controller